File: LeetCode/[M] - 1679. Max Number of K-Sum Pairs/solution.py
class Solution:
    def maxOperations(self, nums: list[int], k: int) -> int:
        hash_count = {}
        for num in nums:
            hash_count[num] = hash_count.get(num, 0) + 1
        res = 0
        for num in nums:
            find = k - num
            if hash_count[num] and hash_count.get(find, 0):
                if hash_count[find] > 1 or num != find:
                    hash_count[find] -= 1
                    hash_count[num] -= 1
                    res += 1
        return res

File: LeetCode/[M] - 1679. Max Number of K-Sum Pairs/solution.go
package main

func maxOperations(nums []int, k int) int {
	hashCnt := make(map[int]int, len(nums)/2)
	for _, num := range nums {
		hashCnt[num] += 1
	}
	res := 0
	for _, num := range nums {
		find := k - num
		if hashCnt[num] != 0 && hashCnt[find] != 0 {
			if hashCnt[find] > 1 || num != find {
				hashCnt[num]--
				hashCnt[find]--
				res++
			}
		}
	}
	return res
}
